pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -60,9 +69,26 @@ def __init__(
             target_column=target_column,
             date_column=date_column,
             channel_columns=channel_columns,
+            channel_prior=channel_prior,
             validate_data=validate_data,
             adstock_max_lag=adstock_max_lag,
         )
+        self._validate_priors()
+
+    def _validate_priors(self) -> None:
+        if self.channel_prior is not None and self.channel_prior.shape != (
+            len(self.channel_columns),
+        ):
+            raise ValueError(
+                f"Channel prior shape {self.channel_prior.shape} does not match the number of channels {len(self.channel_columns)}"
+            )
+
+    def _preprocess_channel_prior(self) -> TensorVariable:
+        return (
+            pm.HalfNormal.dist(sigma=2, shape=len(self.channel_columns))
```

Comment:
@ricardoV94 Is there a nice way the user could simply pass `channel_prior=pm.HalfNormal.dist(sigma=2)` and add the shape internally? That is, we modify the user input and set `pm.HalfNormal.dist(sigma=2, shape=len(self.channel_columns)` as we currently have? Otherwise, the information seems redundant right?

Reply:
Yes with `change_dist_size`, it's used inside `pm.Censored` if you want to see it being used.

Reply:
Amazing! (this is a hidden gem if you ask me ;)

Reply:
There are some gotchas, like you cant resize an intermediate variable after a dependency was created (another variable already depends on it), so we don't advertise it too much.